File: src/main/java/vip/yazilim/libs/springcore/rest/model/RestError.java
package vip.yazilim.libs.springcore.rest.model;

import lombok.Data;

/**
 * @author Emre Sen, 24.07.2019
 * @contact maemresen@yazilim.vip
 */
@Data
public class RestError {

    private int errorCode;
    private String message;

    public RestError(int errorCode, String message) {
        this.errorCode = errorCode;
        this.message = message;
    }
}

File: src/main/kotlin/vip/yazilim/libs/springvip/util/generic/rest/model/RestError.kt
package vip.yazilim.libs.springvip.util.generic.rest.model

/**
 * The model that represents the error
 *
 * @author Emre Sen, 24.07.2019
 * @contact maemresen@yazilim.vip
 */
data class RestError<C>(private val errorCode: Int, private val errorCause: C)
